### DISCUSSION TITLE
Can we use all the deployment types with the jreleaser.yml configuration file?

### DISCUSSION THREAD

**Discussion by TomerFi** [jreleaser/jreleaser#155] [Q&A] [answered]
I would like to use the `maven` plugin for manual builds and the `github` action for automated builds.
While reserving the option to use other deployments of `jreleaser` if needed (e.g. `docker`, `jbang`, `local`).

Is it possible instructing the `maven` plugin and `github` action to load their respective configuration for the relative `jreleaser.yml` file?

**Comment by aalmiray** [comment, marked as answer]
At the moment no, the Maven plugin requires the use of its DSL to define the model. Conversely the CLI tool used by the GitHub Action can only read `.yml`, `.toml`, or `.json` external formats.

You may trigger releases in a GitHub workflow using the Maven plugin, just invoke `mvn jreleaser:full-release`. The `jreleaser/release-action` is just a convenient way to download and setup the CLI.

This being said, you're not the only one that has asked if the Maven plugin can consume external formats, that is certainly doable. The other way round is much harder, as that would require the CLI to be aware of Maven types.

**Reply by aalmiray** [reply]
Just added the feature with https://github.com/jreleaser/jreleaser/issues/157

**Reply by TomerFi** [reply]
Confirmed with early-access.
Thank you very much.
